File: python/calculate_mode.py
from operator import countOf


def calculate_mode(x):
    answer=[]
    compare=0
    myObj=[]
    #print(countOf(x, 4.5))
    for i in x:
        count=countOf(x, i)
        myObj.append([i, count])
    #print(myObj)
    for i in myObj:
        if(compare<=i[1]):
            compare= i[1]
        else:
            continue
    #print(compare)
    for j in myObj:
        if(j[1]== compare):
            answer.append(j[0])
        else:
            continue
    for k in answer:
        if countOf(answer, k)>1:
            answer.remove(k)
    return answer
#print(calculate_mode(["who", "what", "where", "who"]))

File: js/calculateMode.js
exports.calculateMode = (x) => {
    answer=[]
    compare=0
    let myObj={}
    x.forEach(function(i){myObj[i]=((myObj[i]||0)+1)})
    values=(Object.values(myObj))
    keys=(Object.keys(myObj))
    for (i=0;i<values.length;i++){
        if(values[i]>compare){
            compare= values[i]
        }
        else{
            continue
        }
    }
    for(j=0;j<values.length;j++){
        if(values[j]===compare){
            answer.push(keys[j])
        }
    }
    return(answer)
}


//calculateMode([4.5, 0, 0]) 
